src/core/Particles.tsx

```diff
@@ -40,26 +39,62 @@ declare global {
   }
 }
 
+const isFloat32Array = (def: any): def is Float32Array => def && (def as Float32Array).constructor === Float32Array
+
+const expandColor = (v: THREE.Color) => [v.r, v.g, v.b]
+const isVector = (v: any): v is THREE.Vector2 | THREE.Vector3 | THREE.Vector4 =>
+  v instanceof THREE.Vector2 || v instanceof THREE.Vector3 || v instanceof THREE.Vector4
+
+const normalizeVector = (v: any): number[] => {
+  if (Array.isArray(v)) return v
+  else if (isVector(v)) return v.toArray()
+
+  return [v, v, v] as number[]
+}
+
+function usePropAsIsOrAsAttribute<T extends any>(
+  count: number,
+  prop?: T | Float32Array,
+  setDefault?: (v: T) => number
+) {
+  return React.useMemo(() => {
+    if (prop !== undefined) {
+      if (isFloat32Array(prop)) {
+        return prop as Float32Array
+      } else {
+        if (prop instanceof THREE.Color) {
+          const a = Array.from({ length: count * 3 }, () => expandColor(prop)).flat()
+          return Float32Array.from(a)
+        } else if (isVector(prop) || Array.isArray(prop)) {
+          const a = Array.from({ length: count * 3 }, () => normalizeVector(prop)).flat()
+          return Float32Array.from(a)
+        }
+        return Float32Array.from({ length: count }, () => prop as number)
+      }
+    }
+    return Float32Array.from({ length: count }, setDefault!)
+  }, [prop])
+}
+
 export const Particles = React.forwardRef<THREE.Points, Props & PointsProps>(
-  ({ count = 100, speed = 1, opacity = 1, pixelRatio, scale = [1, 1, 1], size = 1, color, ...props }, forwardRef) => {
+  (
+    { noise = 1, count = 100, speed = 1, opacity = 1, pixelRatio, scale = [1, 1, 1], size, color, ...props },
+    forwardRef
+  ) => {
     const matRef = React.useRef<any>()
     const positions = Float32Array.from(
-      Array.from({ length: count }, () => [
-        THREE.MathUtils.randFloatSpread(scale[0]),
-        THREE.MathUtils.randFloatSpread(scale[1]),
-        THREE.MathUtils.randFloatSpread(scale[2]),
-      ]).flat()
+      Array.from({ length: count }, () => scale.map(THREE.MathUtils.randFloatSpread)).flat()
+    )
+
+    const sizes = usePropAsIsOrAsAttribute<number>(count, size, () => Math.random())
+    const opacities = usePropAsIsOrAsAttribute<number>(count, opacity)
+    const speeds = usePropAsIsOrAsAttribute<number>(count, speed)
+    const noises = usePropAsIsOrAsAttribute<typeof noise>(count * 3, noise)
```

Comment:
Is `noise` here the amount of noise influence on each particle?

Reply:
yep

Reply:
https://github.com/pmndrs/drei/pull/857#issuecomment-1094909000

Reply:
gotcha!